tests/unit/test_charm.py

```diff
@@ -344,13 +345,86 @@ def bundle_handler(args: ops.testing.ExecArgs) -> None:
     charm: DiscourseCharm = typing.cast(DiscourseCharm, harness.charm)
 
     email = "sample@email.com"
-    password = "somepassword"  # nosec
     event = MagicMock(spec=ActionEvent)
-    event.params = {
-        "email": email,
-        "password": password,
-    }
-    charm._on_add_admin_user_action(event)
+    event.params = {"email": email}
+    charm._on_promote_user_action(event)
+    assert expected_exec_call_was_made
+
+
+@patch.object(ops.Container, "exec")
+def test_create_user(mock_exec):
+    """
+    arrange: an email
+    act: when the _on_create_user_action method is executed
+    assert: the create user rake command is executed upon failure of the user existence check.
+    """
+    mock_wo = MagicMock(
+        return_value=("", None),
+    )
+    stdout_mock = "CRASH"
+    harness = helpers.start_harness(run_initial_hooks=False)
+    harness.set_can_connect(CONTAINER_NAME, True)
+
+    email = "admin-user@test.internal"
+    expected_exec_call_was_made = False
+
+    def mock_wo_side_effect(*args, **kwargs):  # pylint: disable=unused-argument
```

Comment:
This mocking could be substituted with something like:
```
    harness.handle_exec(
        SERVICE_NAME,
        [f"{DISCOURSE_PATH}/bin/bundle", "exec", "rake", f"users:exists[{email}]"],
        result=ops.testing.ExecResult(exit_code=2, stdout="not found", stderr="")
    )
    ```

Reply:
This has been implemented, much nicer. Thanks